enclave_wrangler/objects_api.py

```diff
@@ -368,46 +369,49 @@ def fetch_cset_and_member_objects(
-                'fail-excessive-items', 'comment': f"Failed after {len(cset['expression_items'])} expression_items."})
-        except Exception as err:
-            insert_fetch_status(fetch_statuses)
-            raise err
+            if handle_issues:
+                insert_fetch_statuses([{'table': 'code_sets', 'primary_key': version_id, 'status_initially':
+                    'fail-excessive-items', 'comment':
+                    f"Failed after {len(cset['expression_items'])} expression_items."}])
+                call_github_action('resolve-fetch-failures-excess-items')
         try:
-            cset['member_items']: List[Dict] = get_concept_set_version_members(version, return_detail='full')
+            cset['member_items']: List[Dict] = get_concept_set_version_members(version_id, return_detail='full')
         except EnclavePaginationLimitErr as err:
             cset['member_items']: List[Dict] = err.args[1]['results_prior_to_error']
-            fetch_statuses.append({'table': 'code_sets', 'primary_key': version, 'status_initially':
-                'fail-excessive-members', 'comment': f"Failed after {len(cset['member_items'])} members."})
-        except Exception as err:
-            insert_fetch_status(fetch_statuses)
-            raise err
-        # todo: consider adding `and cset['expression_items']:` to further avoid false positives, or is the isDraft
-        #  check sufficient? I think it's correct that items are available immediately upon creation, but members are
-        #  not? Can a draft be finalized and have 0 expression items? Another thing to check would be createdAt >1+hours
-        #  as proxy; e.g. 'createdAt': '2023-02-14T15:10:13.921Z',
-        if not cset['member_items'] and not cset['properties']['isDraft']:
-            fetch_statuses.append({'table': 'code_sets', 'primary_key': version, 'status_initially': 'fail-0-members',
-                'comment': f"Fetched 0 members after fetching {len(cset['expression_items'])} items."})
+            if handle_issues:
+                insert_fetch_statuses([{'table': 'code_sets', 'primary_key': version_id, 'status_initially':
+                    'fail-excessive-members', 'comment': f"Failed after {len(cset['member_items'])} members."}])
+                call_github_action('resolve-fetch-failures-excess-items')
+        if not cset['member_items'] and cset['expression_items'] and not cset['properties']['isDraft'] and handle_issues:
+            insert_fetch_statuses([{
+                'table': 'code_sets', 'primary_key': version_id, 'status_initially': 'fail-0-members',
+                'comment': f"Fetched 0 members after fetching {len(cset['expression_items'])} items."}])
+            call_github_action('resolve_fetch_failures_0_members.yml', {'version_id': version_id})
+
         cset_versions_with_concepts.append(cset)
         expression_items.extend(cset['expression_items'])
         member_items.extend(cset['member_items'])
 
-    if insert_fetch_status:
```

Comment:
## Failures handled immediately when happen
@Sigfried You wanted me to change things so that when we find an issue during the refresh, we immediately handle it, rather than waiting until the end of `fetch_cset_and_member_objects()` to do that.

Initially I disagreed with you, because I thought that we'd get several instances where we'd have failures due to >10k expression items, etc, and that we'd have a bunch of failures queued up waiting for us to download a dataset to fix that, and I thought it'd be better to do this all at once.

In reality, that is a very rare issue. So I changed to immediately upon fetch failure.

I suppose I could have the different failure types (excessive items, vs 0 members) handled differently in this way, but it really doesn't matter.